apps/studio/components/layouts/TableEditorLayout/ExportAllRows.progress.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useRef, type ReactNode } from 'react'
+import { toast } from 'sonner'
+import { SonnerProgress } from 'ui'
+
+export const useProgressToasts = () => {
+  const toastIdsRef = useRef(new Map<number, string | number>())
+
+  const startProgressTracker = useCallback(
+    ({
+      id,
+      name,
+      trackPercentage = false,
+    }: {
+      id: number
+      name: string
+      trackPercentage?: boolean
+    }) => {
+      if (toastIdsRef.current.has(id)) return
+
+      if (trackPercentage) {
+        toastIdsRef.current.set(
+          id,
+          toast(<SonnerProgress progress={0} message={`Exporting ${name}...`} />, {
+            closeButton: false,
+            duration: Infinity,
+          })
+        )
+      } else {
+        toastIdsRef.current.set(id, toast.loading(`Exporting ${name}...`))
+      }
+    },
+    []
+  )
+
+  const trackPercentageProgress = useCallback(
+    ({
+      id,
+      name,
+      value,
+      totalRows,
+    }: {
+      id: number
+      name: string
+      value: number
+      totalRows: number
+    }) => {
+      const savedToastId = toastIdsRef.current.get(id)
+
+      const progress = Math.min((value / totalRows) * 100, 100)
+      const newToastId = toast(
+        <SonnerProgress progress={progress} message={`Exporting ${name}...`} />,
```

Comment:
am just wondering if there's a better way we can display the progress here - if the table is really long, it looks as though nothing might be happening since its only a spinner (even if we're constantly paging in the background)